DALib/Drawing/Graphics.cs

```diff
@@ -193,43 +193,50 @@ public static SKImage RenderMap(MapFile map, DataArchive seoDat, DataArchive iaD
     ///     Renders a MapFile, given already extracted information
     /// </summary>
     /// <param name="map">
-    ///     The MapFile to render
+    ///     The <see cref="MapFile"/> to render
     /// </param>
     /// <param name="tiles">
-    ///     A collection of background tiles
+    ///     A <see cref="Tileset"/> representing a collection of background tiles
     /// </param>
     /// <param name="bgPaletteLookup">
-    ///     PaletteLookup for background tiles
+    ///     <see cref="PaletteLookup"/> for background tiles
     /// </param>
     /// <param name="fgPaletteLookup">
-    ///     PaletteLookup for foreground tiles
+    ///     <see cref="PaletteLookup"/> for foreground tiles
     /// </param>
     /// <param name="iaDat">
-    ///     IA archive for reading foreground tile files
+    ///     IA <see cref="DataArchive"/> for reading foreground tile files
+    /// </param>
+    /// <param name="foregroundPadding">
+    ///     The amount of padding to add to the height of the file and beginning rendering position
+    /// </param>
+    /// <param name="cache">
+    ///     A <see cref="MapImageCache"/> that can be reused to share <see cref="SKImageCache{TKey}"/> caches between multiple map renderings.
     /// </param>
     public static SKImage RenderMap(
         MapFile map,
         Tileset tiles,
         PaletteLookup bgPaletteLookup,
         PaletteLookup fgPaletteLookup,
-        DataArchive iaDat)
+        DataArchive iaDat, 
+        int foregroundPadding = 512,
+        MapImageCache? cache = null)
     {
-        const int FOREGROUND_PADDING = 512;
 
```

Comment:
could do something like this...
```suggestion
//only dispose the cache if it was created in this method
var dispose = cache is null;
cache ??= new MapImageCache();

try
{
    ...
} finally 
{
   if (dispose)
       cache.Dispose();
}
```